### PR TITLE
CMR-4103: Added functionality to respect all the temporal ranges in t…

### PR DESCRIPTION
**Pull request by sxu123**
…he collection temporal search.

This PR replaces the old PR (The old branch was messed up) with the fixes that address the PR comments. Also added integration tests.

### CODE REVIEW

File: indexer-app/src/cmr/indexer/data/concepts/collection.clj [hunk 1]
```diff
@@ -49,6 +50,10 @@
       :else
       (errors/internal-error! (str "Unknown spatial representation [" coord-sys "]")))))
 
+(defn- apply-function-to-all-values-in-map [m f]
```
Comment (indiejames) [inline indexer-app/src/cmr/indexer/data/concepts/collection.clj]: Please move the argument declaration below the docstring.

File: indexer-app/src/cmr/indexer/data/concepts/collection.clj [hunk 2]
```diff
@@ -193,8 +206,15 @@
...
         coordinate-system (get-in collection [:SpatialExtent :HorizontalSpatialDomain
-                                              :Geometry :CoordinateSystem])
-        permitted-group-ids (get-coll-permitted-group-ids context provider-id collection)]
+                                                       :Geometry :CoordinateSystem])
+        permitted-group-ids (get-coll-permitted-group-ids context provider-id collection)
+        {:keys [granule-start-date granule-end-date]} (cgac/get-coll-gran-aggregates context concept-id)
+        last-3-days (t/interval (t/minus (tk/now) (t/days 3)) (tk/now))
+        granule-end-date (when-not (and granule-end-date (t/within? last-3-days granule-end-date))
+                           granule-end-date)]
+                           ;; If the granule end date is within the last 3 days we indicate that
```
Comment (indiejames) [inline indexer-app/src/cmr/indexer/data/concepts/collection.clj]: Does this comment refer to the previous line? If so move it above that line, please.

Comment (sxu123) [inline indexer-app/src/cmr/indexer/data/concepts/collection.clj]: It used to be if statement that I copied from another place. Mark wanted this be changed to when-not. but the comment is for "when" case, which I think is useful.

File: indexer-app/src/cmr/indexer/data/concepts/collection.clj (hunk 1 above)
Comment (sxu123) [inline indexer-app/src/cmr/indexer/data/concepts/collection.clj]: Done.

File: indexer-app/src/cmr/indexer/data/concepts/collection.clj (hunk 2 above)
Comment (sxu123) [inline indexer-app/src/cmr/indexer/data/concepts/collection.clj]: done.

**Pull request merged**